epimodel/tasks.py

```diff
@@ -0,0 +1,502 @@
+        default="1D",
+        description="Subsampling of imported data (see Pandas `df.resample`)",
+    )
+
+
+CONFIG = Configuration()
+
+
+def default_from_config(task_name: str, param_name: str) -> dict:
+    return dict(section=task_name, name=param_name)
+
+
+def _prefix_cfg(filename: str, par_attrib="input_directory") -> str:
+    return (Path(getattr(CONFIG, par_attrib)) / filename).as_posix()
+
+
+class RegionsFile(luigi.ExternalTask):
+    regions = luigi.Parameter(default=_prefix_cfg("regions.csv"))
+
+    def output(self):
+        return luigi.LocalTarget(self.regions)
+
+
+class GleamRegions(luigi.ExternalTask):
+    gleams = luigi.Parameter(default=_prefix_cfg("regions-gleam.csv"))
+
+    def output(self):
+        return luigi.LocalTarget(self.gleams)
+
+
+class RegionsAggregates(luigi.ExternalTask):
+    aggregates = luigi.Parameter(default=_prefix_cfg("regions-agg.yaml"))
+
+    def output(self):
+        return luigi.LocalTarget(self.aggregates)
+
+
+@inherits(RegionsFile, GleamRegions, RegionsAggregates)
+class RegionsDatasetTask(luigi.Task):
+    regions_dataset: str = luigi.Parameter(
+        default=_prefix_cfg("region_dataset.pk", "output_directory")
+    )
+
+    def run(self):
+        regions = self.input()["region_file"].path
+        gleams = self.input()["gleam_regions"].path
+        aggregates = self.input()["aggregates"].path
+        rds = RegionDataset.load(regions, gleams, aggregates)
+        algorithms.estimate_missing_populations(rds)
+        with open(self.regions_dataset, "wb") as ofile:
+            dill.dump(rds, ofile)
+
+    def output(self):
+        return luigi.LocalTarget(self.regions_dataset)
+
+    def requires(self):
+        return {
+            "region_file": self.clone(RegionsFile),
+            "gleam_regions": self.clone(GleamRegions),
+            "aggregates": self.clone(RegionsAggregates),
+        }
+
+    @staticmethod
+    def load_dilled_rds(path: str):
+        with open(path, "rb") as ifile:
+            return dill.load(ifile)
+
+
+class JohnsHopkins(luigi.Task):
+    _output_directory: str = luigi.Parameter(default=CONFIG.output_directory)
+    hopkins_output: str = luigi.Parameter(
+        config_path=default_from_config("JohnsHopkins", "hopkins_output")
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._full_output_path = Path(self._output_directory) / self.hopkins_output
+
+    def run(self):
+        rds = RegionsDatasetTask.load_dilled_rds(self.input().path)
+        csse = imports.import_johns_hopkins(rds)
+        csse.to_csv(self._full_output_path)
+        logger.info(
+            f"Saved CSSE to {self._full_output_path}, last day is {csse.index.get_level_values(1).max()}"
+        )
+
+    def output(self):
+        return luigi.LocalTarget(self._full_output_path)
+
+    def requires(self):
+        return RegionsDatasetTask()
+
+
+class UpdateForetold(luigi.Task):
```

Comment:
Maybe here us an `luigi.DateParameter(default=datetime.date.today())` as well, only make it run less frequently? Maybe there is a way to check freshness otherwise?

Reply:
Here is a random example of complete method override: https://gist.github.com/demoray/b503c887518941d264b0